pymc3/models/base.py

```diff
@@ -0,0 +1,69 @@
+from ..model import modelcontext
+from collections import OrderedDict
+
+
+class UserModel(object):
+    """Base class for model specification
+    It is supposed to be used by pymc3 contributors
+    for simplifying usage of common bayesian models
+    """
+
+    def __init__(self, name):
+        # name should be used as prefix for
+        # all variables specified within a model
+        self.vars = OrderedDict()
+        self.name = name
+
+    @property
+    def model(self):
+        """Shortcut to model"""
+        return modelcontext(None)
+
+    def new_var(self, name, dist, data=None, test_val=None):
```

Comment:
I also thought about a method like [this](https://github.com/pymc-devs/pymc3/pull/1525/files#diff-f85ae0776100e0962d318aaad0c5340fR88) for base class, but i'm not sure if it will be good for readability. Here it exists to avoid code duplication